bobtimus/src/amounts.rs

```diff
@@ -0,0 +1,104 @@
+use anyhow::Result;
+use elements_fun::bitcoin::Amount;
+use rust_decimal::{prelude::ToPrimitive, Decimal};
+use serde::{Deserialize, Serialize};
+use std::{convert::TryFrom, fmt::Debug, ops::Mul};
+
+/// Prices at which 1 L-BTC will be traded, in L-USDt.
+///
+/// - The `ask` represents the minimum price for which we are willing to sell 1 L-BTC.
+/// - The `bid` represents the maximum price we are willing pay for 1 L-BTC.
+#[derive(Debug, Clone, Copy, Serialize)]
+pub struct Rate {
+    pub ask: Decimal,
```

Comment:
think the type of this field should be a `LiquidBtc`? Semantically, it is the amount you pay/receive for 1 L-BTC.

Reply:
I suppose you mean `LiquidUsdt`. I wanted to use `Decimal` to mimic what was done in Nectar, but maybe there is no need here.

Reply:
I think I've improved things with regards to `Rate` and `LiquidUsdt`. I'll push a commit.

Reply:
https://github.com/comit-network/droplet/pull/22/commits/558a2bc7743eda741757b1942e6059709c2d8e6b

Reply:
I've force pushed (maybe too soon), but all the other changes are simple.